PolyEngine/Scripts/ProjectTool.py

```diff
@@ -0,0 +1,120 @@
+        setattr(namespace, self.dest, ActionType.UPDATE)
+        setattr(namespace, self.metavar, values[0])
+
+# Functions
+# Create project function
+
+# Setup:
+# $PROJECT_NAME$ - name of the project
+# $PROJECT_PATH$ - relative path of the project (game) directory (relative to root project dir)
+# $GAME_CLASS_NAME$ - name of the game class
+# $ENGINE_DIR$ - absolute path to engine root directory
+
+def ReplaceTagsInFile(fileToSearch, tagsAndValues):
+    print('Replacing tags in', fileToSearch)
+    for (tag, val) in tagsAndValues:
+        print('\tchanging', tag, 'to', val)
+
+    with fileinput.FileInput(fileToSearch, inplace=True) as file:
+        for line in file:
+            for (tag, val) in tagsAndValues:
+                line = line.replace(tag, val)
+            print(line, end='')
+
+def RunCmake(path, buildDir):
+    # Run cmake update (using undocumented parameters that work for some reason, src: http://cprieto.com/posts/2016/10/cmake-out-of-source-build.html)
+    if os.name == 'nt':
+        os.system(r'cmake -G "Visual Studio 14 2015 Win64" -H' + path + ' -B' + buildDir)
+    else:
+        os.system(r'cmake -H' + path + ' -B' + buildDir)
+
+def CreateProject(name, path, enginePath):
+    print('Creating project', name, 'in', path, 'with engine at', enginePath)
+
+    if os.path.exists(path):
+        raise ValueError('Path', path, 'already exists. Cannot create project there!')
+    
+    path = os.path.abspath(path)
+
+
+    os.makedirs(path)
+    os.makedirs(path + os.sep + 'Build')
+
+    projectPath = path + os.sep + name
+    os.makedirs(projectPath)
+    projectResourcesPath = projectPath + os.sep + 'Res'
+    os.makedirs(projectResourcesPath)
+    projectSourcesPath = projectPath + os.sep + 'Src'
+    os.makedirs(projectSourcesPath)
+
+    scriptsDataPath = enginePath + os.sep + 'Scripts'
+    shutil.copy(scriptsDataPath + os.sep + 'Game.hpp.in', projectSourcesPath + os.sep + 'Game.hpp')
+    shutil.copy(scriptsDataPath + os.sep + 'Game.cpp.in', projectSourcesPath + os.sep + 'Game.cpp')
+    shutil.copy(scriptsDataPath + os.sep + 'Sln-CMakeLists.txt.in', path + os.sep + 'CMakeLists.txt')
+    shutil.copy(scriptsDataPath + os.sep + 'Proj-CMakeLists.txt.in', projectPath + os.sep + 'CMakeLists.txt')
+
+    ReplaceTagsInFile(projectSourcesPath + os.sep + 'Game.hpp', [('$GAME_CLASS_NAME$', 'Game')])
+    ReplaceTagsInFile(projectSourcesPath + os.sep + 'Game.cpp', [('$GAME_CLASS_NAME$', 'Game')])
+    ReplaceTagsInFile(path + os.sep + 'CMakeLists.txt', [('$PROJECT_PATH$', name), ('$ENGINE_DIR$', os.path.abspath(enginePath).replace('\\', '/'))])
+    ReplaceTagsInFile(projectPath + os.sep + 'CMakeLists.txt', [('$PROJECT_NAME$', name)])
+
+    RunCmake(path, path + '/Build')
+
+def UpdateProject(path, enginePath):
+    print('Updating project at', path, 'with engine at', enginePath)
+    RunCmake(path, path + '/Build')
+
+#################### SCRIPT START ####################
```

Comment:
You should use `if __name__ == "__main__":` to indicate where this script actually starts